ams/server/src/main/java/com/netease/arctic/server/manager/FlinkOptimizerContainer.java

```diff
@@ -307,27 +360,41 @@ private <T> T fetchCommandOutput(Process exec, Function<String, T> commandReader
 
   @Override
   public void releaseOptimizer(Resource resource) {
-    String releaseCommand;
-    switch (target) {
-      case YARN_APPLICATION:
-      case YARN_PER_JOB:
-        releaseCommand = buildReleaseYarnCommand(resource);
-        break;
-      case KUBERNETES_APPLICATION:
-        releaseCommand = buildReleaseKubernetesCommand(resource);
-        break;
-      default:
-        throw new IllegalStateException("Unsupported running target: " + target.getValue());
-    }
+    if (target.runByFlinkRestClient()) {
+      Preconditions.checkArgument(
+          resource.getProperties().containsKey(SESSION_CLUSTER_JOB_ID),
+          "Cannot find {} from optimizer start up stats",
+          SESSION_CLUSTER_JOB_ID);
+      String jobId = resource.getProperties().get(SESSION_CLUSTER_JOB_ID);
+      try (RestClusterClient<String> restClusterClient =
+          FlinkClientUtil.getRestClusterClient(loadFlinkConfig())) {
+        Acknowledge ignore = restClusterClient.cancel(JobID.fromHexString(jobId)).get();
```

Comment:
It is not needed here because `Acknowledge` is just a generic acknowledgement message. return result in flink client. If an exception occurs, Exception will be thrown directly here. However, may need to add a timeout detection here.